src/shared/dbgutil/elfreader.cpp

```diff
@@ -543,6 +550,40 @@ ElfReader::GetBuildId(BYTE* buffer, ULONG bufferSize, PULONG pBuildSize)
 
 #ifdef HOST_UNIX
 
+bool
+ElfReader::GetBuildIdFromSectionHeader(uint64_t baseAddress, BYTE* buffer, ULONG bufferSize, PULONG pBuildSize)
+{
+    Elf_Ehdr ehdr;
+    if (!ReadHeader(baseAddress, ehdr)) {
+        return false;
+    }
+    if (ehdr.e_shoff == 0 || ehdr.e_shnum <= 0) {
+        return false;
+    }
+    Elf_Shdr* shdrAddr = reinterpret_cast<Elf_Shdr*>(baseAddress + ehdr.e_shoff);
+    for (int sectionIndex = 0; sectionIndex < ehdr.e_shnum; sectionIndex++, shdrAddr++)
+    {
+        Elf_Shdr sh;
+        if (!ReadMemory(shdrAddr, &sh, sizeof(sh))) {
```

Comment:
Should this be &shdrAddr[sectionIndex]?

Reply:
No, because the shdrAddr++ in the for increments the address by the sizeof(ELF_Shdr).  sectionIndex is used just to count the sections.